config/solr.yml

```diff
@@ -4,4 +4,4 @@ development:
 test:
   url: http://127.0.0.1:<%= ENV['SOLR_TEST_PORT'] || 8985 %>/solr/hydra-test
 production:
-  url: http://127.0.0.1:8983/solr/californica
+  url: <%= ENV['RAILS_ENV'] == 'production' ? ENV.fetch('SOLR_URL') : ENV['SOLR_URL'] %>
```

Comment:
It feels like  `ENV['RAILS_ENV'] == 'production'`  is redundant under a `production:` section - how would this line ever be run under the state were `ENV['RAILS_ENV'] == 'development'`?

This syntax would make more sense in a `default:` section. 

I'm also more used to using .fetch if I want to provide a default when the value isn't set - e.g. I think of these two as being equivalent, but maybe that's not right:
```
ENV.fetch('SOLR_URL', 'http://127.0.0.1:8983/solr/californica')
ENV['SOLR_URL'] || 'http://127.0.0.1:8983/solr/californica'

Reply:
Sorry - I didn't see you'r comment about the use of fetch - so I'd be a fan of this to raise the error in production.   
```
production:
  url: <%= ENV.fetch('SOLR_URL') %>
```

Reply:
Your example is what I tried first. This will raise in *any* environment unless `SOLR_URL` is set, and therefore renders the default settings in `development` and `test` useless.

Reply:
Specifically: the full file is parsed in every environment, so the raise from `#fetch` happens regardless of environment, even if the actual variable will never be referenced.

----

Another option would be to add an initializer that explicitly checks for variables we want to ensure are present. I initially rejected this as "yet another place we need to know ENV variable names", but it's looking more attractive the more I think about it.

If we want to go down this road, I'd suggest we merge this without the last commit, and open a new PR for the initializer.

Reply:
Shoot - I didn't know about the full file being parsed :(  

I'd kinda lean toward keeping the config files as simple as we can - i.e. without this conversation for context, the syntax ` url: <%= ENV['RAILS_ENV'] == 'production' ? ENV.fetch('SOLR_URL') : ENV['SOLR_URL'] %>` seems a little opaque.

I'd slightly lean toward for keeping things simple and dropping the requirement "raise an error if the value isn't set" and just go with
```
production:
  url: <%= ENV.fetch['SOLR_URL'] %>
```

If we want to add a ticket for "raise an error when key settings are missing", we could explore options later.  

If that sounds good, go for it.  If you'd like to keep the current syntax to raise the config error, I'm good with that too.